pkg/controllers/disruption/consolidation_test.go

```diff
@@ -28,6 +28,7 @@ import (
 	. "github.com/onsi/ginkgo/v2"
```

Comment:
Maybe you'll cover this with the helper testing, but I'm curious if we should have coverage over checking cases where we have nodes getting marked for deletion for some other reason and these should be counted against the disruption budget.

Reply:
We only do MarkForDeletion within the disruption controller, so I don't think we need to worry about this. Even if there were, I don't think it'd be unreasonable to expect this to be considered as part of the budget.

Reply:
Ah I thought you meant if there were other controllers calling `statenode.MarkForDeletion`. I've added tests that incorporate if a node is deleting, which would then be updated in ClusterState to be seen as MarkedForDeletion.